### ISSUE TITLE
Ignoring of request headers

### ISSUE THREAD

**Issue by jackkitley** [rakutentech/laravel-request-docs#67]
If i use the url and try authenticate via the docs i always get a error as it was the CSRF token which is not needed for the sanctum api request.

If i export to postman and click the button to post to auth then i get the response i am looking for..
![Screenshot 2022-04-26 at 15 41 31](https://user-images.githubusercontent.com/3243014/165314077-a545e34e-874b-467c-bb3a-3a74b3581200.png)
![Screenshot 2022-04-26 at 15 44 09](https://user-images.githubusercontent.com/3243014/165314092-8b6bfe96-9a60-489a-a9f5-8ef8da62e550.png)

If i try postman i get the correct response:
<img width="862" alt="Screenshot 2022-04-26 at 15 45 51" src="https://user-images.githubusercontent.com/3243014/165314152-066a2f2a-2f02-4cd3-b25d-7c9425066017.png">

**Comment by kevincobain2000** [comment]
Hi, thanks for reporting.
Let me have a look. 
If I remove adding X-CSRF-TOKEN to headers when it's value is empty, would work for you?

**Comment by kevincobain2000** [comment]
Like this: https://github.com/rakutentech/laravel-request-docs/pull/68

**Comment by jackkitley** [comment]
This is the request that is not working from the documents. I caught the request in console:

curl 'http://localhost/api/auth' \
  -H 'Accept: application/json, text/plain, */*' \
  -H 'Accept-Language: en-GB,en-US;q=0.9,en;q=0.8' \
  -H 'Authorization: Bearer' \
  -H 'Cache-Control: no-cache' \
  -H 'Connection: keep-alive' \
  -H 'Content-Type: application/json;charset=UTF-8' \
  -H 'Cookie: JSESSIONID.2b84989e=node07l6xggrciay8jx7295segy967.node0; JSESSIONID.5353d309=node01l7d23qeq46bnqv7uhe64qtfl2.node0; jenkins-timestamper-offset=-7200000; JSESSIONID.e0139d4a=node01jc68r3nruauq1qryxpvtsuy359.node0; screenResolution=1920x1080; JSESSIONID.c93846e1=node07tt6op3njuqywuwa128iem0p0.node0; JSESSIONID.25ec0c08=node026o0go5nquq1daizxvtuj3552.node0; JSESSIONID.afa366ea=node088cymhiju4gx19jtkaaaaktcn4.node0; JSESSIONID.b2d12f37=node0h6phxoodtlw21892ftjp56vrs0.node0; JSESSIONID.8df212d3=node01a7m3arx63ovt12qkzl4m9z4430.node0; JSESSIONID.9a3073d2=node061k4j5olaxo6lphqxghvnkt70.node0; JSESSIONID.5b57156e=node01nd9y4s0s8zpj19alodss0fy510.node0; JSESSIONID.ed350b34=node01olt5dl2iulbfytkij40v3b6m2.node0; JSESSIONID.d79d91e4=node019bleri6n83s21t63gh4sbqvbr0.node0; XDEBUG_PROFILE=; PHPSESSID=90300e5a16bdf041b4fa3a54453ef9bd; _ga=GA1.1.975793485.1650558215' \
  -H 'Origin: http://localhost' \
  -H 'Pragma: no-cache' \
  -H 'Referer: http://localhost/request_doc' \
  -H 'Sec-Fetch-Dest: empty' \
  -H 'Sec-Fetch-Mode: cors' \
  -H 'Sec-Fetch-Site: same-origin' \
  -H 'Sec-GPC: 1' \
  -H 'User-Agent: Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/100.0.4896.79 Safari/537.36' \
  -H 'X-CSRF-TOKEN: ' \
  -H 'X-Request-LRD: lrd' \
  --data-raw '{"email":"","password":""}' \
  --compressed

This is working from postman extract:

curl --location --request POST 'http://localhost/api/auth' \
--header 'Content-Type: application/json' \
--header 'Accept: application/json' \
--data-raw '{
  "email": "",
  "password": ""
}'

**Comment by jackkitley** [comment]
If i take out

  -H 'Origin: http://localhost' \
  -H 'Referer: http://localhost/request_doc' \

then it works. Wonder if its Cors

**Comment by kevincobain2000** [comment]
Interesting. I don't think it is cors issue, but trusthosts or trust proxies issues, which you try to change it inside your project.

```
class TrustProxies extends Middleware
{
    /**
     * The trusted proxies for this application.
     *
     * @var array|string|null
     */
    protected $proxies = ['127.0.0.1', 'localhost'];
```

Not sure, I ll try and reproduce on my local

**Comment by jackkitley** [comment]
> Interesting. I don't think it is cors issue, but trusthosts or trust proxies issues, which you try to change it inside your project.
> 
> ```
> class TrustProxies extends Middleware
> {
>     /**
>      * The trusted proxies for this application.
>      *
>      * @var array|string|null
>      */
>     protected $proxies = ['127.0.0.1', 'localhost'];
> ```
> 
> Not sure, I ll try and reproduce on my local

Ok, not a cors issue.

{
  "message": "CSRF token mismatch.",
  "exception": "Symfony\\Component\\HttpKernel\\Exception\\HttpException",
  "file": "/var/www/html/vendor/laravel/framework/src/Illuminate/Foundation/Exceptions/Handler.php",
  "line": 357,
  "trace": [
    {
      "file": "/var/www/html/vendor/laravel/framework/src/Illuminate/Foundation/Exceptions/Handler.php",
      "line": 331,
      "function": "prepareException",
      "class": "Illuminate\\Foundation\\Exceptions\\Handler",
      "type": "->"
    },

This is with the docs. My api works with sanctum

**Comment by jackkitley** [comment]
I generated the docs again and now its working. very strange. thanks

**Comment by kevincobain2000** [comment]
Thanks for the update buddy @jackkitley

**Comment by jackkitley** [comment]
@kevincobain2000 Happening again.
![Screenshot 2022-04-28 at 13 16 48](https://user-images.githubusercontent.com/3243014/165740729-e52913ae-c7c6-4118-a607-1d3d0adc49d6.png)

**Comment by jackkitley** [comment]
also getting this in console:
![Screenshot 2022-04-28 at 13 24 32](https://user-images.githubusercontent.com/3243014/165741898-c6b42566-fbf6-4c59-839a-1119016cd424.png)

**Comment by jackkitley** [comment]
I replicated the issue which is a strange one for the API docs. When you authenticate via the website and get a session and then go back to the api docs and click run then it all works again? very strange.
![Screenshot 2022-04-28 at 13 42 41](https://user-images.githubusercontent.com/3243014/165744822-9908e664-1b97-4dd5-afbe-b8ce4a35c723.png)

**Comment by kevincobain2000** [comment]
Well, that is the correct behavior. The API docs are just like any other page. Once you are logged into your domain, the cookies for session etc are set by Laravel. And when the API call is made, the api is able to authenticate the user.

**Comment by jackkitley** [comment]
> Well, that is the correct behavior. The API docs are just like any other page. Once you are logged into your domain, the cookies for session etc are set by Laravel. And when the API call is made, the api is able to authenticate the user.

Well in postman i can call the endpoints without setting a cookie/session. I should be able to call the endpoints openly. 
Im not using CRSF, im using sanctum token based api calls.

**Comment by jackkitley** [comment]
Just an update. If i modify the verify CSRF middleware to exclude the checks for /api/* then it works. But its still strange that these requests are coming through as web.
![Screenshot 2022-04-29 at 08 45 42](https://user-images.githubusercontent.com/3243014/165896674-4db08347-1d89-4438-8854-e3228325fc7a.png)

**Comment by kevincobain2000** [comment]
Fixed in #124
